tests/MSBuildDeviceIntegration/Tests/DebuggingTest.cs

```diff
@@ -251,36 +251,60 @@ public override void OnCreate ()
 				/* embedAssemblies */    true,
 				/* fastDevType */        "Assemblies",
 				/* allowDeltaInstall */  false,
+				/* user */		 "Owner",
 			},
 			new object[] {
 				/* embedAssemblies */    false,
 				/* fastDevType */        "Assemblies",
 				/* allowDeltaInstall */  false,
+				/* user */		 "Owner",
 			},
 			new object[] {
 				/* embedAssemblies */    false,
 				/* fastDevType */        "Assemblies",
 				/* allowDeltaInstall */  true,
+				/* user */		 "Owner",
 			},
 			new object[] {
 				/* embedAssemblies */    false,
 				/* fastDevType */        "Assemblies:Dexes",
 				/* allowDeltaInstall */  false,
+				/* user */		 "Owner",
 			},
 			new object[] {
 				/* embedAssemblies */    false,
 				/* fastDevType */        "Assemblies:Dexes",
 				/* allowDeltaInstall */  true,
+				/* user */		 "Owner",
+			},
+			new object[] {
+				/* embedAssemblies */    true,
+				/* fastDevType */        "Assemblies",
+				/* allowDeltaInstall */  false,
+				/* user */		 DeviceTest.GuestUserName,
+			},
+			new object[] {
+				/* embedAssemblies */    false,
+				/* fastDevType */        "Assemblies",
+				/* allowDeltaInstall */  false,
+				/* user */		 DeviceTest.GuestUserName,
 			},
 		};
 #pragma warning restore 414
 
 		[Test, Category ("SmokeTests"), Category ("Debugger")]
 		[TestCaseSource (nameof(DebuggerTestCases))]
-		public void ApplicationRunsWithDebuggerAndBreaks (bool embedAssemblies, string fastDevType, bool allowDeltaInstall)
+		public void ApplicationRunsWithDebuggerAndBreaks (bool embedAssemblies, string fastDevType, bool allowDeltaInstall, string username)
 		{
 			AssertCommercialBuild ();
 			AssertHasDevices ();
+
+			int userId = GetUserId (username);
+			if (SwitchUser (username)) {
+				WaitFor (5);
+				ClickButton ("", "android:id/button1", "Yes continue");
+			}
```

Comment:
Maybe instead of `Owner` we should just allow `username` to be blank. So we know these tests will work if you don't set `$(AndroidDeviceUserId)` at all?

Reply:
good idea